dotCMS/src/main/java/com/dotcms/content/elasticsearch/business/ESMappingAPIImpl.java

```diff
@@ -514,11 +533,14 @@ else if (f.getFieldType().equals(ESMappingConstants.FIELD_ELASTIC_TYPE_DATE)) {
 
 				} else {
 					if (f.getFieldContentlet().startsWith(ESMappingConstants.FIELD_ELASTIC_TYPE_BOOLEAN)) {
-						m.put(st.getVelocityVarName() + "." + f.getVelocityVarName(), valueObj.toString());
+						m.put(st.getVelocityVarName() + "." + f.getVelocityVarName(), valueObj);
```

Comment:
you should use StringBuilder

Reply:
not sure if necessary, doesn't the compiler turn it into stringbuilder in this case?

Reply:
Ah depends on the implementation, so might not be always the case. Agreed on the StringBuilder then @jdotcms @nollymar